### ISSUE TITLE
[HOLD #12885] [$500] [Performance] Comment send button not switching state instantenously 

### ISSUE THREAD

**Issue by hannojg** [Expensify/App#11697]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## What performance issue do we need to solve?

It's a minor thing that I noticed during creating https://github.com/Expensify/App/pull/11684.
When typing text into the comment report composer, the send button takes a while to switch between the active or disabled state:

## What is the impact of this on end-users?

https://user-images.githubusercontent.com/16821682/194876067-c2da7bb8-3b5f-43de-98c7-1f2fe58a8679.mp4

## List any benchmarks that show the severity of the issue

You can track e.g. the JS thread to see how it freezes/locks when entering and deleting the comment.

## Proposed solution (if any)

I found that the lines after are problematic:
https://github.com/Expensify/App/blob/e7d6b52843e06d0cb20b171ddab72b17d572e9c2/src/pages/home/report/ReportActionCompose.js#L387-L405

Either we defer that code, or we could investigate why these lines cause such a big performance impact.

## List any benchmarks after implementing the changes to show impacts of the proposed solution (if any)

https://user-images.githubusercontent.com/16821682/194876295-997eaf45-4bb0-49b3-afdf-d7a2191e3f97.mp4

When simply removing the lines you can see that the JS thread doesn't lock.

## Platform:
Where is this issue occurring?

- [ ] Web
- [x] iOS
- [x] Android
- [ ] Desktop App
- [ ] Mobile Web

**Version Number:** Latest main branch, but its 1.2.12-2
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

**Comment by johnmlee101** [comment]
I notice a small delay on my own device, so I think it would be nice if we could speed htis up

**Comment by hannojg** [comment]
No this issue isn't addressed by the mentioned PR. This issue was created while I was working on the PR and found this issue and wanted to keep track of it.

**Comment by johnmlee101** [comment]
Ah okay cool, then I think this is a good thing to fix and let the assigned engineer determine if the existing proposal meets the requirements!

**Comment by deetergp** [comment]
While you're in that part of the code, it'd be nice to figure out what's causing that lag and seeing if we can speed it up rather than simply deferring it. If we find that we can't speed it up, then deferring it would be acceptable.

**Comment by deetergp** [comment]
Not overdue

**Comment by deetergp** [comment]
I re-applied the External label so we'd get a CM/BugZero assignee

@hannojg Did you ever find out where the slowdown was coming from?

**Comment by hannojg** [comment]
Not yet. Most likely from calls to onyx which in general often cause a lock of the JS thread. Will look again into that issue more in depth in the coming days

**Comment by vladamx** [comment]
This needs a whole suite of changes. It is not one thing. A lot of that work is either done across other issues or in drafts.

cc @marcaaron

**Comment by maddylewis** [comment]
hi sorry for the delay here - still getting up to speed on BZ stuff. i will get the job posted to Upwork and post the link to this issue.

**Comment by maddylewis** [comment]
job posted here - https://www.upwork.com/jobs/~019259c7134745d072

**Comment by khan-shazid** [comment]
@hannojg , 
I am going through this issue and testing it on iPhone 13 emulator. But could not find the delay. Should I use any specific device to test this one? or a test account with which I can generate this issue? I am running it from latest main branch.

**Comment by hannojg** [comment]
just try with a weaker device if you have one available 😊

**Comment by deetergp** [comment]
Please correct me if I am wrong, but I believe we are still awaiting updated proposals for this one.

**Comment by maddylewis** [comment]
@deetergp - should i be doubling the price on this one? or, should i hold off until we have an agreed-upon proposal to do that?

**Comment by maddylewis** [comment]
doubled the price on this one - https://www.upwork.com/jobs/~019259c7134745d072

**Comment by mananjadhav** [comment]
@maddylewis Still waiting on proposals.

**Comment by maddylewis** [comment]
i will double the price on this one today and update here once that's complete.

**Comment by s77rt** [comment]
Not reproducible from my end.

**Comment by hannojg** [comment]
I think it's in relation to the amount of reports/chats that you have. The more you have the worse the issue. You can test with a high traffic account

**Comment by iwiznia** [comment]
FWIW I can't reproduce either on my Pixel 6.

**Comment by iwiznia** [comment]
I think we should close this. It is a performance issue on slow/old phones and we have https://github.com/Expensify/App/issues/12885 to improve android performance

**Comment by iwiznia** [comment]
@hannojg (or anyone that could reproduce this) on what real devices have we experienced this?

**Comment by mountiny** [comment]
@hannojg Also could you reproduce this issue with the fixes you had already been testing for [this issue](https://github.com/Expensify/App/issues/12885)?

As discussed [in slack](https://expensify.slack.com/archives/C01SKUP7QR0/p1669045814616379) (internal), I will put this on hold for the new performance issue resolution, which looks promising and might fix this as well.

**Comment by iwiznia** [comment]
Still, let's document in which real device this issue is seen.

**Comment by tjferriss** [comment]
Coming here from the [weekly bug update chore](https://stackoverflow.com/c/expensify/questions/8208) and I tend to agree with Ioni that if this is not reproducible on a real device than let's close it.

**Comment by hannojg** [comment]
I still have this on my todo list and will try to reproduce it on devices in the release app.

**Comment by maddylewis** [comment]
@hannojg - i'm going to close this out for now. if it ends up being reproducible, feel free to reopen with your findings.
